mysql-test/suite/rocksdb/t/rocksdb_mrr.test

```diff
@@ -0,0 +1,389 @@
+
+--echo # This will not use MRR, as one of the ranges is a non-singlepoint:
+explain select * from t2 where pk in (1,2,3,4,5) or pk between 6 and 8;
+
+create table t3 (
+  pk1 int,
+  pk2 int,
+  col1 int,
+  filler char(32),
+  primary key(pk1, pk2),
+  key(col1)
+) engine=rocksdb;
+
+insert into t3 select a,a, a,a from t1;
+set global rocksdb_force_flush_memtable_now=1;
+
+--echo # This must not use BKA as it uses key prefix:
+explain 
+select * from t3,t0 where t3.pk1=t0.a;
+
+--echo # This will use BKA as it uses full key:
+explain 
+select * from t3,t0 where t3.pk1=t0.a and t3.pk2=t0.a+1;
+
+--echo # This will use MRR as it uses full key:
+explain select * from t3 where pk1 in (1,2,3,4,5) and pk2 in (3,4,5);
+
+--echo # This will not use MRR as there are non-singlepoint ranges:
+explain select * from t3 where (pk1 in (1,2,4,5) and pk2 in (4,5)) or pk1=3;
+
+--echo # Due to HA_MRR_SUPPORT_SORTED, the following will use MRR, and not require filesort:
+explain select * from t3 where pk1 in (1,2,3,4,5) and pk2 in (3,4,5) order by pk1, pk2;
+
+#--echo #
+#--echo # More basic tests
+#--echo #
+
+--echo # Test for lookups returning nothing
+delete from t2 where pk in (3,5,7,9,10) or pk between 100 and 200;
+
+explain
+select * from t2,t0 where t2.pk=t0.a;
+select * from t2,t0 where t2.pk=t0.a;
+
+
+--echo # Test for a scan returning nothing at all:
+explain
+select * from t2,t0 where t2.pk=t0.a+500100;
+select * from t2,t0 where t2.pk=t0.a+500100;
+
+--echo # Make the scan finish before it has returned all the data:
+select * from t2,t0 where t2.pk=t0.a limit 1;
+
+--echo # Re-execute MRR scan in a subquery:
+create table t4(a int primary key);
+insert into t4 values (4),(6),(8);
+
+--replace_column 9 #
+explain
+select a ,
+       (select concat(t0.a,'-',t2.pk) 
+        from t2,t0 
+        where t2.pk=t0.a and t2.col1>=t4.a
+        limit 1) as SUBQ
+from t4;
+
+select a ,
+       (select concat(t0.a,'-',t2.pk) 
+        from t2,t0 
+        where t2.pk=t0.a and t2.col1>=t4.a
+        limit 1) as SUBQ
+from t4;
+
+--echo #
+--echo # MRR/BKA on secondary keys
+--echo #
+
+--echo # This won't use BKA because it's an index-only scan:
+explain
+select t3.col1 from t3,t0 where t3.col1=t0.a;
+
+--echo # This will use BKA: 
+explain
+select * from t3,t0 where t3.col1=t0.a;
+
+select * from t3,t0 where t3.col1=t0.a;
+
+--echo # Now, run the query and check the counters
+create temporary table t11 as
+select * from information_schema.global_status where variable_name like 'rocksdb_num%';
+
+select * from t3,t0 where t3.col1=t0.a;
+
+create temporary table t12 as
+select * from information_schema.global_status where variable_name like 'rocksdb_num%';
+
+select 
+  variable_name, 
+  t12.variable_value - t11.variable_value as DIFF 
+from 
+  t11 join t12 using (VARIABLE_NAME)
+having
+  DIFF>0;
+
+drop table t11,t12;
+
+--echo #
+--echo # Check the userstat counters
+--echo #
+show variables like '%user_table_stat%';
```

Comment:
This isn't a useful variable to check. This controls the population of the `information_schema.user_table_statistics` table (not table_statistics). We can probably just remove this.

Reply:
Removed